diffkemp/output.py

```diff
@@ -1,10 +1,107 @@
+"""
+Contains logic for creating output of DiffKemp compare command.
+
+Specifically, the YAML file is created here which contains
+information about found differences and definitions of symbols
+which are located in the differences.
+"""
 from diffkemp.semdiff.caching import ComparisonGraph
 from diffkemp.utils import get_end_line, EndLineNotFound
 from diffkemp.semdiff.result import Result
 import os
 import yaml
 
 
+class MacroDefinitions:
+    """Class for extracting information about macro definitions."""
+    def __init__(self, snapshot_dir_old, snapshot_dir_new, result):
+        self.old_dir = snapshot_dir_old
+        self.new_dir = snapshot_dir_new
+        self.result = result
+        # Information from call stacks for extracting defs for differing macros
+        # - list of tuples (vertex name, differing/last macro name).
+        self.diff_macro_infos = set()
+        # Definitions in YamlOutput format.
+        self.definitions = {}
+
+    def extract_from_result(self, called_res, compared_fun):
+        """
+        Extracts information about macro definitions from result of called
+        entity.
+        :param called_res: Result of called entity.
+        :param compared_fun: Name of compared function.
+        """
+        # Note: Definitions of differing/last macro is not possible to extract
+        # directly from the call stacks, we will now extract for it only
+        # name of the macro and function/vertex name, the definition will be
+        # extracted later in `_add_defs_for_differing` method.
+        if called_res.first.callstack:
+            macro_defs, diff_macro_info = called_res.first.callstack \
+                .get_macro_defs(compared_fun)
+            self._add_definitions(macro_defs, "old")
```

Comment:
Hm, I am not a big fan of the `"old", "new", "kind"` string literal hard-coding. It would be nice if we could use a string enum for this instead but I am not sure which python versions diffkemp aims to support (we are still supporting something old, aren't we?). If that is the case, feel free to ignore this.

Reply:
Thanks for the review. I replaced the `old`/`new` with boolean parameter `is_old`. For the `kind` I used enum which inherits from `str, Enum` (this should be available from Python 3.4), I could use `StrEnum` but it is supported from Python 3.11 and as you said I am also not sure which Python versions we want to support.